### PR TITLE
fix for Jenkins log URL

### PR DESCRIPTION
**Pull request by hrishin**
- added a new ENV var JENKINS_ROOT_URL to openshift-deployment.yml
fix for #64

### CODE REVIEW

**Comment by pradeepto** [comment]
cc @vpavlin @hrishin @maxandersen We need a review for this PR.

File: apps/jenkins/src/main/fabric8/openshift-deployment.yml [hunk 1]
```diff
@@ -76,6 +76,8 @@ spec:
             secretKeyRef:
               name: jenkins-recommender-api-token
               key: token
+        - name: JENKINS_ROOT_URL
+          value: "https://jenkins.openshift.io"
```
Comment (vpavlin) [inline apps/jenkins/src/main/fabric8/openshift-deployment.yml]: We need this value to be different for prod-preview (https://jenkins.prod-preview.openshift.io) and production  (https://jenkins.openshift.io) - how will we achieve this?

**Comment by fabric8cd** [comment]
PR now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?jenkinsVersion=PR-v7d7674a-2&mavenRepo=https://nexus.cd.test.fabric8.io/content/repositories/staging) and click the update tenant button

File: apps/jenkins/src/main/fabric8/openshift-deployment.yml (hunk 1 above)
Comment (hrishin) [reply apps/jenkins/src/main/fabric8/openshift-deployment.yml]: How about `Value: "${URL_PLACEHOLDER}"` and CI/CD or deployment tools can substitute this value for target environment.

@vpavlin @rupalibehera @aslakknutsen

**Comment by hrishin** [comment]
How about Value: "${URL_PLACEHOLDER}" and CI/CD or deployment tools (oc CLI) can substitute this value for the target environment, basically templating
Or
leverage on Configmaps?

@vpavlin @rupalibehera @aslakknutsen

**Comment by fabric8cd** [comment]
PR now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?jenkinsVersion=PR-v857976f-3&mavenRepo=https://nexus.cd.test.fabric8.io/content/repositories/staging) and click the update tenant button

File: apps/jenkins/src/main/fabric8/openshift-deployment.yml [hunk 2]
```diff
@@ -76,6 +76,8 @@ spec:
             secretKeyRef:
               name: jenkins-recommender-api-token
               key: token
+        - name: JENKINS_ROOT_URL
+          value: "${JENKINS_ROOT_URL}"
```
Comment (hrishin) [inline apps/jenkins/src/main/fabric8/openshift-deployment.yml]: As @vpavlin said prod-preview has (https://jenkins.prod-preview.openshift.io) and production (https://jenkins.openshift.io) - URL's?

To achieve this following could be possible options:
1) Templating: To use `${JENKINS_ROOT_URL}` template variable CI/CD or deployment tools can substitute this value
e.g. 
`oc <> -P JENKINS_ROOT_URL=https://jenkins.openshift.io`
`mvn fabric8:resource -DJENKINS_ROOT_URL=https://jenkins.openshift.io`

2) Use Configmaps?

 @rupalibehera @aslakknutsen

Comment (aslakknutsen) [reply apps/jenkins/src/main/fabric8/openshift-deployment.yml]: @hrishin 1 and 2 are the same. The 'deployment tool' in question here is the Tenant Service. There is no ConfigMap outside of what Tenant Service sets up for you. If you want to set it up via the "jenkins" ConfigMap(https://github.com/fabric8-services/fabric8-tenant-jenkins/blob/master/apps/jenkins/src/main/fabric8/openshift-cm.yml) or directly on the DeploymentConfig is same same for me. But the 'real' value will be in the Tenant Service ConfigMap.

File: apps/jenkins/src/main/fabric8/openshift-deployment.yml [hunk 3]
```diff
@@ -76,6 +76,8 @@ spec:
             secretKeyRef:
               name: jenkins-recommender-api-token
               key: token
+        - name: JENKINS_ROOT_URL
```
Comment (aslakknutsen) [inline apps/jenkins/src/main/fabric8/openshift-deployment.yml]: Add value to Tempalte https://github.com/fabric8-services/fabric8-tenant-jenkins/blob/master/apps/jenkins/src/main/fabric8/template.yml

**Comment by fabric8cd** [comment]
PR now available for testing: Launch in [OpenShift.io](https://openshift.io/_profile/_tenant?jenkinsVersion=PR-va62e91a-4&mavenRepo=https://nexus.cd.test.fabric8.io/content/repositories/staging) and click the update tenant button

**Pull request merged**